### DISCUSSION TITLE
Question about the pm values datatype.

### DISCUSSION THREAD

**Discussion by LGnahore** [avaldebe/PMserial#27] [Q&A]
Hello, i have been working with this library recently and have noticed that the datatype of the pm values is uint16_t.
Is there any simple way that i can get those values as floats or doubles or does it require big adjustments?

(If it is relevant, I am using an ESP8266 and a PMS7003.)

**Comment by avaldebe** [comment]
you probably want to cast integer to a float, as in this [example](https://www.arduino.cc/reference/en/language/variables/data-types/float)

``` c++
int x;
int y;
float z;

x = 1;
y = x / 2;          // y now contains 0, ints can't hold fractions
z = (float)x / 2.
```
